src/dl_typelib_write_c_header.cpp

```diff
@@ -222,43 +223,50 @@ static void dl_context_write_c_header_end( dl_binary_writer* writer, const char*
 	dl_binary_writer_write_string_fmt( writer, "#endif // __DL_AUTOGEN_HEADER_%s_INCLUDED\n\n", module_name_uppercase );
 }
 
-static void dl_context_write_c_header_typeids( dl_binary_writer* writer, dl_ctx_t ctx )
+template <class T>
+using unique_malloc_ptr = std::unique_ptr<T, decltype( &free )>;
+
+static dl_error_t dl_context_write_c_header_typeids( dl_binary_writer* writer, dl_ctx_t ctx )
 {
 	dl_binary_writer_write_string_fmt( writer, "//----------------------------------------------\n"
 											   "//                   TYPEID:s                   \n"
 											   "//----------------------------------------------\n\n" );
 
 	dl_type_context_info_t ctx_info;
-	dl_reflect_context_info( ctx, &ctx_info );
+	dl_error_t err = dl_reflect_context_info( ctx, &ctx_info );
+	if( err != DL_ERROR_OK )
+		return err;
 
 	if(ctx_info.num_enums != 0)
 	{
-		dl_enum_info_t* enum_infos = (dl_enum_info_t*)malloc( ctx_info.num_enums * sizeof( dl_enum_info_t ) );
-		dl_reflect_loaded_enums(ctx, enum_infos, ctx_info.num_enums);
+		unique_malloc_ptr<dl_enum_info_t[]> enum_infos( (dl_enum_info_t*)malloc( ctx_info.num_enums * sizeof(dl_enum_info_t) ), free );
```

Comment:
Did not see this before? If you want automatic cleanup I would prefer a DL_DEFERED-macro

Reply:
It is also not prefixed with dl_

Reply:
Do you mean a special "run this lambda in the destructor" define?

Reply:
Something like this yes https://www.gingerbill.org/article/2015/08/19/defer-in-cpp/